pkg/clustermanager/cluster_manager.go

```diff
@@ -277,7 +277,13 @@ func (c *ClusterManager) CreateWorkloadCluster(ctx context.Context, managementCl
 	logger.V(3).Info("Waiting for control plane to be ready")
 	err = c.clusterClient.WaitForControlPlaneReady(ctx, managementCluster, c.controlPlaneWaitTimeout.String(), workloadCluster.Name)
 	if err != nil {
-		return nil, fmt.Errorf("waiting for workload cluster control plane to be ready: %v", err)
+		err = c.Retrier.Retry(
+			func() error {
+				workloadCluster.KubeconfigFile, err = c.generateWorkloadKubeconfig(ctx, workloadCluster.Name, managementCluster, provider)
+				return err
+			},
+		)
+		return workloadCluster, fmt.Errorf("waiting for workload cluster control plane to be ready: %v", err)
```

Comment:
we can just `return nil, fmt.Errorf()`.
also in this case, you are overriding the `err` from line 279, with err generated from line 280. There is no way to indicate the `WaitForControlPlaneReady` error then.

Reply:
If kubeconfig is set, this would allow the support bundle to be created for workload cluster, otherwise if you return nil it won't

Reply:
Yeah, I made that change to return a nil instead of workloadcluster. 
And right, I probably need to re-name the `err` to something like `error` to distinguish the two?

Reply:
I am little bit confused here. If we don't return workLoadCluster at line 286, how can we get the kubeconfig? It seems like that kubeconfigFile is one of the attributes of workLoadCluster, right?

Reply:
i see, make sense. the overriding error msg still valid right

Reply:
In order to make this work though, we need to also add this line in the err block in this file right? https://github.com/aws/eks-anywhere/blob/249f1abdf2a3c740ef3e05fdbf4031353c7460bb/pkg/workflows/create.go#L219

Reply:
I would vote to find an alternative to this.

This method now returns:
* (valid cluster, nil)
* (nil, error)
* (valid cluster, error)
* (invalid cluster without kubeconfig, error)

It's a bit confusing, it requires the caller to know about the internal implementation and the returned values are not easy to handle.

I would start by listing the scenarios where `WaitForControlPlaneReady` might fail but we might still be able to generate the kubeconfig.

Reply:
I think it does make sense though right? If we get an error, the workflow will go through the error path, and if not, it won't. All we want from this method is whatever form of cluster is ready, and we already have those existing validations for generating the support bundle in the diagnostic workflow. But do you feel that we should instead be splitting this up more to only return a combination of `nil, error`?

Reply:
I think we need a better definition of what this method does. Does it just send the request to create the cluster? Does it wait until the kubeconfig is ready? Does it wait until the CP is ready?

Based on the previous implementation, a CP ready was the minimum we needed to consider the cluster created. Now it seems we think we could have a kubeconfig without the control plane being ready. But a kubeconfig without a control plane doesn't do anything. So that makes me think we need to analyze a bit more that scenario and list where `WaitForControlPlaneReady` might fail but we might still be able to generate the kubeconfig. I think that will point us into the right direction, which probably involves making some more granular condition checks (more granular than `WaitForControlPlaneReady`).

Reply:
> I think we need a better definition of what this method does. Does it just send the request to create the cluster? Does it wait until the kubeconfig is ready? Does it wait until the CP is ready?
> 
> Based on the previous implementation, a CP ready was the minimum we needed to consider the cluster created. Now it seems we think we could have a kubeconfig without the control plane being ready. But a kubeconfig without a control plane doesn't do anything. So that makes me think we need to analyze a bit more that scenario and list where `WaitForControlPlaneReady` might fail but we might still be able to generate the kubeconfig. I think that will point us into the right direction, which probably involves making some more granular condition checks (more granular than `WaitForControlPlaneReady`).

Just want to clarify that we do need a kubeconfig without the control plane being ready. The fact got from multiple tests has verified that, without the cp being ready, the kubeconfig can help support bundle to diagnose cluster. What my unsolicited thought about this condition is, as long as control plane is created, kubeconig should be valuable.

Reply:
I understand that. But what I think we need are the condition under which the control plane might not be ready but the kubeconfig is available and useful.